python/py-graph-tool/Portfile

```
@@ -70,9 +72,25 @@ if {${name} ne ${subport}} {
                     port:py${python.version}-numpy \
                     port:py${python.version}-scipy
 
+    depends_lib-append \
+                    port:py${python.version}-matplotlib
+
     # graph-tool relies on Boost.Python, so make sure it is installed.
     require_active_variants boost[boost::version_nodot] python${python.version}
 
+    # remove after this issue is fixed:
+    # https://git.skewed.de/count0/graph-tool/-/issues/761
+    post-patch {
+        fs-traverse f ${worksrcpath}/src {
+            if { [file isfile ${f}]
+                && [regexp ".(cc|hh)" [file extension ${f}]] } {
+                reinplace -q -E \
+                    "s|:\[\[:space:]]+public\[\[:space:]]+std::unary_function<.+,.+>||" \
```

Does upstream agree that this is an appropriate fix?

Since you are not interpolating Tcl variables or procedure calls, you can eliminate the need for escaping by using `{}` quoting characters instead of `""`:

```suggestion
                    {s|:[[:space:]]+public[[:space:]]+std::unary_function<.+,.+>||} \
```

---

Thanks, done.

---

Upstream @count0 is aware of the fix at https://git.skewed.de/count0/graph-tool/-/issues/761. The `unary_function` is unnecessary and has been removed from C++17, so this will certainly work. I'll defer to Tiago how he'd like to address this and have noted in the Portfile that this Portfile-based fix should be removed after the issue is resolved.